### DISCUSSION TITLE
Crazyflie can not take off randomly.

### DISCUSSION THREAD

**Discussion by Jool99** [bitcraze/discussions#1177] [Q&A] [answered]
Hi! We are experiencing that the crazyflie 2.1 with flow deck v2 sometimes flips at take off like in [this video](https://drive.google.com/file/d/1x-UFRqJVlWGCwdG35YWIm7I1Vz6-VCFH/view?usp=sharing). It seems the crazyflie sometimes enters a "state" where it does this repeatedly. 

The only things that seem to make it exit this state are:
- Waiting for at least a few minutes.
- Detaching and reattaching the flow deck (but we are unsure if it is just the time being passed).

It SEEMS the crazyflie is more prone to enter this state:
- When the AI Deck is attached.
- After crashing (or it crashed because it entered this state).
- If much work is done at the beginning of appMain().
- If the flow deck is straight but the pins are tilted.
- The larger 350 mAh batteries are used.

We have tried and ruled out the following in terms of software:
- Restarting.
- Reflashing.
- Standard or our custom firmware.
- Take off from cfclient, crtpCommanderHighLevelTakeoff() from custom firmware app or commanderSetSetpoint() from custom firmware app.

We have tried and ruled out the following in terms of hardware:
- Propeller and flow deck orientation.
- Flow deck attached straight.
- Switching arm, motor and propeller M1 with M3 and M2 with M4 (it flipped in roughly the same direction).
- Tried different drone bodies.
- Tried different flow decks.
- Tried different batteries.
- New propellers.
- Cfclient propeller test. It has flipped and succeeded with both passed and failed tests. Now with 47-17 propellers, all tests fail but it seems unrelated.
- Battery position (when it works, the battery can be half off the body and when it does not work, centering the battery does not help).

All help is appreciated. Thanks in advance!

**Comment by knmcguire** [comment]
Hi! Thanks for this list of things you tried before

 Some more things that you can try perhaps:
- Checking the propeller direction, but unlikely since it was able to fly before: https://www.bitcraze.io/documentation/tutorials/getting-started-with-crazyflie-2-x/#attach-the-propellers
- Try to put the flowdeck on an area with more texture. Perhaps the position estimate had diverged quite a bit. I recommend you to read this blogpost for tips and tricks https://www.bitcraze.io/2023/11/go-with-the-flow-relative-positioning-with-the-flow-deck/
- It might also be that the ground is very static, which can confuse the imu. Make sure to look if the attitude is diverging before take of

Let us know if any of these helps

**Comment by Jool99** [comment]
Thanks for your answer!

* We have checked the propeller directions.
* Not position nor attitude seem to be diverging before take off. Have plotted logs of a take off and crash for reference. See attached image.

![CrashGraphs](https://github.com/bitcraze/discussions/assets/57872258/4b37e1c6-0e45-4fd3-bd33-d1722b6320f0)

**Comment by knmcguire** [comment, marked as answer]
Have you tried perhaps to replace the motors one by one with a new one. Each crazyflie should come with a spare motor. I don't see this your list of tried tactics but please correct me if I'm wrong.

**Reply by Jool99** [reply]
One motor with a crooked axis and at least one more worn was definitely part of the problem. Thank you!

**Reply by knmcguire** [reply]
alright good to know! thanks for the update. Could you select 'mark as answer' then please?
